src/C1T2X_OBU.py

```diff
@@ -137,7 +141,20 @@ def VANET_listening_thread():
 			c1t2x_logger.debug("Received %s from VANET", pkt)
 			if pkt:
 				if not parseVANETPacket:
-					sendLAN(pkt[0])
+					# Check if ack or payload
+					data = pkt[0].decode('utf-8')
+					if data == "1":  # Ack
+						with mutex:
+							wait_for_ack = False
+						c1t2x_logger.info("Received ack")
+					elif data == previous_packet_received:  # Duplicate message received, so just resend ack
+						sendVANET(ack)
+						c1t2x_logger.info("Received duplicate message, resending ack")
+					else:  # New message received, forward it to LAN and send ack
+						sendVANET(ack)
+						sendLAN(pkt[0])
+						previous_packet_received = pkt[0]
```

Comment:
It feels slightly awkward to drop sending ACKs if we receive a new message, but it's probably fine because I don't think we expect to get another message before this one gets through.

Reply:
That's a good point but yeah we shouldn't received another message because each component sends one message before getting a response from the other.